src/Microsoft.Azure.ServiceBus/Core/MessageSender.cs

```diff
@@ -90,6 +92,40 @@ public class MessageSender : ClientEntity, IMessageSender
             }
         }
 
+        private async Task<Message> ProcessMessage(Message message)
+        {
+            if (this.RegisteredPlugins == null)
+            {
+                return message;
+            }
+
+            var processedMessage = message;
+            foreach (var plugin in this.RegisteredPlugins)
+            {
+                try
+                {
+                    processedMessage = await plugin.BeforeMessageSend(message).ConfigureAwait(false);
+                }
+                catch (Exception ex) when (ex is ServiceBusPluginException pluginException && pluginException.ShouldCompleteOperation)
```

Comment:
>pluginException.ShouldCompleteOperation [](start = 94, length = 39)

What are your opinions on exposing this on the plugin? Who should decide whether this is an exception that impacts or not? - User or the plugin developer?